pkg/cmd/pulumi/packagecmd/package_publish_sdk_test.go

```diff
@@ -0,0 +1,204 @@
+			name:           "current version equal to latest",
+			currentVersion: "1.0.0",
+			npmOutput:      "1.0.0",
+			npmStderr:      "",
+			npmError:       false,
+			expectedTag:    "latest",
+		},
+		{
+			name:           "current version less than latest - backport",
+			currentVersion: "1.0.0",
+			npmOutput:      "2.0.0",
+			npmStderr:      "",
+			npmError:       false,
+			expectedTag:    "backport",
+		},
+		{
+			name:           "patch version less than latest - backport",
+			currentVersion: "1.0.0",
+			npmOutput:      "1.0.1",
+			npmStderr:      "",
+			npmError:       false,
+			expectedTag:    "backport",
+		},
+		{
+			name:           "minor version less than latest - backport",
+			currentVersion: "1.0.0",
+			npmOutput:      "1.1.0",
+			npmStderr:      "",
+			npmError:       false,
+			expectedTag:    "backport",
+		},
+		{
+			name:           "patch version greater than latest",
+			currentVersion: "1.0.1",
+			npmOutput:      "1.0.0",
+			npmStderr:      "",
+			npmError:       false,
+			expectedTag:    "latest",
+		},
+		{
+			name:           "invalid current version",
+			currentVersion: "not-a-version",
+			npmOutput:      "1.0.0",
+			npmStderr:      "",
+			npmError:       false,
+			expectedErr:    "failed to parse current version",
+		},
+		{
+			name:           "invalid npm version output",
+			currentVersion: "1.0.0",
+			npmOutput:      "not-a-version",
+			npmStderr:      "",
+			npmError:       false,
+			expectedErr:    "failed to parse latest version",
+		},
+		{
+			name:           "non-404 error - should fail",
+			currentVersion: "1.0.0",
+			npmOutput:      "",
+			npmStderr:      "npm error network timeout",
+			npmError:       true,
+			expectedErr:    "failed to get latest version from npm",
+		},
+		{
+			name:           "error with output but no 404 - should fail",
+			currentVersion: "1.0.0",
+			npmOutput:      "some output",
+			npmStderr:      "npm error authentication failed",
+			npmError:       true,
+			expectedErr:    "failed to get latest version from npm",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Since we can't easily mock exec.Command, we'll test the core logic
+			// by simulating what the function should do with the given inputs
+			currentVer, parseErr := semver.ParseTolerant(strings.TrimSpace(tt.currentVersion))
+
+			if parseErr != nil {
+				assert.Contains(t, tt.expectedErr, "failed to parse current version")
+				return
+			}
+			// Simulate the npm command result
+			if tt.npmError {
+				// Check if it's a 404 error (package not found)
+				if tt.npmOutput == "" && strings.Contains(strings.ToLower(tt.npmStderr), "404") {
+					// Package doesn't exist - should return "latest"
+					assert.Equal(t, "latest", tt.expectedTag)
+					return
+				}
+				// Other errors should fail
+				if tt.expectedErr != "" {
+					assert.Contains(t, tt.expectedErr, "failed to get latest version from npm")
+				}
+				return
+			}
```

Comment:
This doesn't seem like a very useful test.  It's gonna get out of sync with the real code, and it's just testing that the test code is good. If we want to test this, we should factor out a function that takes the result of the `exec.Command` as parameters, and contains the core logic of parsing that.